### PR TITLE
Yahoo broker

### PR DESCRIPTION
**Pull request by tfukaza**

### CODE REVIEW

File: harvest/algo.py [hunk 1]
```diff
@@ -375,7 +375,7 @@ def get_price(self, symbol: str=None) -> float:
         if symbol == None:
             symbol = self.watch[0]
         if len(symbol) <= 6:
-            return self.trader.storage.load(symbol, self.trader.interval)['close']
+            return self.trader.storage.load(symbol, self.trader.interval)[symbol]['close'][-1]
```
Comment (shershah010) [inline harvest/algo.py:378]: I feel like having to index a dictionary for the `symbol` when you specify the symbol in the function is bad code. Storage should keep the code as an single level index.

File: harvest/broker/yahoo.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        super().__init__()
+
+    def setup(self, watch: List[str], interval, trader=None, trader_main=None):
+        self.watch_stock = []
+        self.watch_ticker = {}
+
+        if interval not in self.interval_list:
+            raise Exception(f'Invalid interval {interval}')
+        for s in watch:
+            if s[0] == '@':
```
Comment (shershah010) [inline harvest/broker/yahoo.py:29]: use the is_crypto function in the utils file

File: harvest/algo.py (hunk 1 above)
Comment (tfukaza) [reply harvest/algo.py:378]: I think I'll make this an issue around v0.3 or something, for now it'll do 😅

**Comment by shershah010** [comment]
I think some previous changes have broken our tests. I'll look into them more tomorrow but I think some places still call `storage._aggregate` instead of `storage.aggregate`. Also for that function it breaks if the target interval in not in the storage dictionary which is very likely.

**Comment by shershah010** [comment]
Also this call is incorrect: trader.py line 126 `self.storage.store(s, i, self.streamer.fetch_price_history(s, i))` 

`fetch_price_history` taken in a start and end datetime.

**Comment by tfukaza** [comment]
> Also this call is incorrect: trader.py line 126 `self.storage.store(s, i, self.streamer.fetch_price_history(s, i))`
> 
> `fetch_price_history` taken in a start and end datetime.

`fetch_price_history` interface was updated so if dates are not specified, it will automatically set `start` to 1/1/1970 and `end` to the current date.

**Pull request merged**